### DISCUSSION TITLE
Restoring previous version of a page erases custom scripts and styles

### DISCUSSION THREAD

**Discussion by jtstuedle** [requarks/wiki#3760] [Error / Bug Report]
When you restore a page from a previous version, custom styles and scripts are overwritten/removed.

1) Create a new page
2) Add any custom styles or scripts to the page properties
3) Save the page
4) Make an update to the page content
5) Save the page
6) Go into the history and restore the first version (where you initially added the custom style or script in step 2)
7) Check for the existence of the custom Styles & Scripts

In my tests, all custom styles/scripts are blank. Anyone else able to reproduce?

**Comment by vincentruan** [comment]
From the table "pageHistory", I have found that the filed "extra" was blank, this may be a bug.

**Comment by jvh95** [comment]
I had the same problem today and I had to re-do a lot of code... 😭

**Comment by GDelavy** [comment]
This issue is still ongoing as of 2.5.286.

**Comment by jaredbrogan** [comment]
I've submitted another issue that relates to tags, specifically.
* https://github.com/requarks/wiki/discussions/5615

**Comment by yourjelly** [comment]
Yes, I've reproduced this and also lost a lot of custom scripts and css in the process. I don't know where the custom scripts or styles are saved (they definitely are not saved with the page files either markdown or html. 

In addition to restoring page history, restoring content from a 'dump to disk' also destroys all scripts and css from all pages.

Ideally scripts and css changes would be tracked in the page history and be saved along side the page files.

**Comment by aarongover-phl** [comment]
I wish I could have a developer from the Wiki.js team review this because I'm having this exact issue. It's very annoying.

**Comment by MTalon63** [comment]
Can we please finally get a developer, at least mention it on the wiki?
I would have hopped it would back up all of my data, instead it purged all per page styles and scripts.
